include/deal.II/grid/grid_in.h

```diff
@@ -791,6 +791,30 @@ class GridIn
   const std::map<std::string, Vector<double>> &
   get_cell_data() const;
 
+  /**
+   * Return a map containing point data associated with the elements of an
+   * external vtk format mesh imported using read_vtk().
+   * The format of the returned map is as
+   * follows:
+   * - std::string stores the name of the point data (identifier) as specified
+   * in the external mesh
+   * - Vector<double> stores value for the given identifier in each vertex.
+   * To access the value, use point_data[name_field][global_vertex_index].
+   *
+   * For example, if the vtk mesh contains point data "Pressure" defined on
+   * vertices, then `point_data["Pressure"][0]` provides the pressure defined at
+   * vertex ID '0', which corresponds to index 0 of the vector. The length of
+   * the vector in `point_data["Pressure"]` equals the number of vertices in the
+   * coarse mesh.
+   * The length of the returned vector equals the number of vertices in the
+   * coarse mesh also in case of vector data. For example, if the vtk mesh
+   * contains point data "Velocity", defined at vertices, then
+   * `point_data["Velocity"][0]`provides the three components of the velocity
+   * field at vertex ID '0'.
+   */
+  const std::map<std::string, Vector<double>> &
+  get_point_data() const;
```

Comment:
Could I suggest calling this function `get_vertex_data()`, so that we use neutral language rather than the specific term used by VTK, and then in the documentation of this function make clear that (i) at the moment, this function is only implemented for the VTK reader, and (ii) that in the language of VTK, this corresponds to "point data"?

I think we've long intended to make this sort of data reader available also for other formats. It would be nice if we used function names that are not tied to one specific file format's language.